storage/database/shardedPersister_test.go

```diff
@@ -0,0 +1,285 @@
+		benchmarkPersisterOneKey(b, entries, key, createShardedPersister(b))
+	})
+}
+
+func BenchmarkPersisterOneKey8mil(b *testing.B) {
+	entries, keys := generateKeys(8000000)
+	randIndex := randInt(0, len(keys)-1)
+	key := []byte(keys[randIndex])
+
+	b.Run("persister", func(t *testing.B) {
+		benchmarkPersisterOneKey(b, entries, key, createPersister(b))
+	})
+
+	b.Run("sharded persister", func(t *testing.B) {
+		benchmarkPersisterOneKey(b, entries, key, createShardedPersister(b))
+	})
+}
+
+func BenchmarkPersister1milAllKeys(b *testing.B) {
+	entries, keys := generateKeys(1000000)
+
+	b.Run("persister", func(t *testing.B) {
+		benchmarkPersister(b, entries, createPersister(b), keys)
+	})
+
+	b.Run("sharded persister", func(t *testing.B) {
+		benchmarkPersister(b, entries, createShardedPersister(b), keys)
+	})
+}
+
+func BenchmarkPersister4milAllKeys(b *testing.B) {
+	entries, keys := generateKeys(4000000)
+
+	b.Run("persister", func(t *testing.B) {
+		benchmarkPersister(b, entries, createPersister(b), keys)
+	})
+
+	b.Run("sharded persister", func(t *testing.B) {
+		benchmarkPersister(b, entries, createShardedPersister(b), keys)
+	})
+}
+
+func BenchmarkPersister8milAllKeys(b *testing.B) {
+	entries, keys := generateKeys(4000000)
+
+	b.Run("persister", func(t *testing.B) {
+		benchmarkPersister(b, entries, createPersister(b), keys)
+	})
+
+	b.Run("sharded persister", func(t *testing.B) {
+		benchmarkPersister(b, entries, createShardedPersister(b), keys)
+	})
+}
+
+func BenchmarkPersister1mil10kKeys(b *testing.B) {
+	entries, keys := generateKeys(1000000)
+
+	b.Run("persister", func(t *testing.B) {
+		benchmarkPersister(b, entries, createPersister(b), keys[0:10000])
+	})
+
+	b.Run("sharded persister", func(t *testing.B) {
+		benchmarkPersister(b, entries, createShardedPersister(b), keys[0:10000])
+	})
+}
+
+func BenchmarkPersister1mil100kKeys(b *testing.B) {
+	entries, keys := generateKeys(1000000)
+
+	b.Run("persister", func(t *testing.B) {
+		benchmarkPersister(b, entries, createPersister(b), keys[0:100000])
+	})
+
+	b.Run("sharded persister", func(t *testing.B) {
+		benchmarkPersister(b, entries, createShardedPersister(b), keys[0:100000])
+	})
+}
+
+func BenchmarkPersister10mil(b *testing.B) {
+	entries, keys := generateKeys(10000000)
+
+	b.Run("persister", func(t *testing.B) {
+		benchmarkPersister(b, entries, createPersister(b), keys)
+	})
+
+	b.Run("sharded persister", func(t *testing.B) {
+		benchmarkPersister(b, entries, createShardedPersister(b), keys)
+	})
+}
+
+func createPersister(b *testing.B) storage.Persister {
+	dir := b.TempDir()
+	db, _ := NewLevelDB(dir, 2, 1000, 10)
+
+	return db
+}
+
+func createShardedPersister(b *testing.B) storage.Persister {
+	shardCoordinator, _ := sharding.NewMultiShardCoordinator(3, 0)
+
+	dir := b.TempDir()
+	db, _ := NewShardedPersister(dir, 2, 1000, 10, shardCoordinator)
+
+	return db
+}
+
+func benchmarkPersisterOneKey(b *testing.B, keys map[string][]byte, key []byte, db storage.Persister) {
```

Comment:
This does not benchmark the time needed for storage load. For every get, the key will be found in cache.

Reply:
Maybe you could first create the dbs with the needed data, and then each Get should use another db. Also valid for all the other benchmarks.

Reply:
done